zulipterminal/ui_tools/views.py

```diff
@@ -1441,3 +1441,90 @@ def keypress(self, size: urwid_Size, key: str) -> str:
             )
             return key
         return super().keypress(size, key)
+
+
+class FullMsgView(urwid.Frame):
+    def __init__(self, controller: Any, message: Message,
+                 message_links: 'OrderedDict[str, Tuple[str, int, bool]]',
+                 time_mentions: List[Tuple[str, str]],
+                 title: str) -> None:
+        self.title = title
+        self.controller = controller
+        self.message = message
+        self.message_links = message_links
+        self.time_mentions = time_mentions
+
+        msg_box = MessageBox(message, controller.model, None)
+        if not msg_box.need_recipient_header():
+            msg_box.widget_list.insert(0, msg_box.top_header_bar(msg_box))
+
+        # Setting message header
+        header = msg_box.widget_list[:2]
+        header = header + [urwid.Divider()]
+
+        # Fetch and store raw and formatted message
+        _content_formatted = msg_box.widget_list[2].original_widget[-1]
+
+        response = controller.model.client.get_raw_message(message['id'])
+        _content_raw = 'Raw content unavailable :('
+        if response['result'] == 'success':
+            _content_raw = response['raw_content']
+
+        _content_raw_w = urwid.Columns([
+            (1, urwid.Text('')),
+            urwid.Text(_content_raw),
+        ])
+
+        tab_list = [
+            {'label': 'Formatted content', 'content': _content_formatted},
+            {'label': 'Raw content',       'content': _content_raw_w},
+        ]
+
+        self.tab_labels = [tab['label'] for tab in tab_list]
+        self.tab_contents = [tab['content'] for tab in tab_list]
+        self.tab_focus = 0  # Initial tab
+
+        self._render_body()  # Responsible render the body
+
+        super().__init__(self.contents['body'][0], (urwid.Pile(header, None)),
+                         focus_part='body')
+
+        max_cols, max_rows = controller.maximum_popup_dimensions()
+
+        self.width = min(max_cols, self.contents['body'][0].width)
```

Comment:
This seems to generate narrow popups, even when the base window is quite wide?
For large messages we want to display as much as we can :)